src/script/lua_api/l_areastore.cpp

```diff
@@ -0,0 +1,420 @@
+
+	LuaAreaStore *o = checkobject(L, 1);
+	AreaStore *ast = o->as;
+
+	v3s16 pos = check_v3s16(L, 2);
+
+	bool borders = true;
+	bool data = false;
+	get_data_and_border_flags(L, 3, &borders, &data);
+
+	std::vector<Area *> res;
+	
+	ast->getAreasForPos(&res, pos);
+	push_areas(L, res, borders, data);
+
+	return 1;
+}
+
+// get_areas_in_area(area, accept_overlap, borders, data)
+int LuaAreaStore::l_get_areas_in_area(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+
+	LuaAreaStore *o = checkobject(L, 1);
+	AreaStore *ast = o->as;
+
+	luaL_checktype(L, 2, LUA_TTABLE);
+	lua_getfield(L, 2, "min");
+	v3s16 minedge = check_v3s16(L, -1);
+	lua_pop(L, 1);
+
+	lua_getfield(L, 2, "max");
+	v3s16 maxedge = check_v3s16(L, -1);
+	lua_pop(L, 1);
+
+	bool borders = true;
+	bool data = false;
+	bool accept_overlap = false;
+	if (lua_isboolean(L, 3)) {
+		accept_overlap = lua_toboolean(L, 3);
+		get_data_and_border_flags(L, 4, &borders, &data);
+	}
+	std::vector<Area *> res;
+	
+	ast->getAreasInArea(&res, minedge, maxedge, accept_overlap);
+	push_areas(L, res, borders, data);
+
+	return 1;
+}
+
+// insert_area(area)
+int LuaAreaStore::l_insert_area(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+
+	LuaAreaStore *o = checkobject(L, 1);
+	AreaStore *ast = o->as;
+
+	Area a;
+
+	luaL_checktype(L, 2, LUA_TTABLE);
+	lua_getfield(L, 2, "min");
+	a.minedge = check_v3s16(L, -1);
+	lua_pop(L, 1);
+
+	lua_getfield(L, 2, "max");
+	a.maxedge = check_v3s16(L, -1);
+	lua_pop(L, 1);
+
+	a.extremifyEdges();
+	a.id = ast->getFreeId(a.minedge, a.maxedge);
+
+	if (a.id != 0) {
+		lua_getfield(L, 2, "data");
+		size_t d_len;
+		const char *data = luaL_checklstring(L,-1, &d_len);
+		lua_pop(L, 1);
+
+		a.data = new char[d_len];
+		a.datalen = d_len;
+
+		memcpy(a.data, data, d_len);
+
+		ast->insertArea(a);
+
+		a.data = NULL; // To not delete our array after a is destructed
+
+		lua_pushnumber(L, a.id);
+	} else {
+		// couldn't get free id
+		lua_pushboolean(L, false);
+	}
+	return 1;
+}
+
+// reserve(count)
+int LuaAreaStore::l_reserve(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+
+	LuaAreaStore *o = checkobject(L, 1);
+	AreaStore *ast = o->as;
+
+	size_t count = luaL_checknumber(L, 2);
+	ast->reserve(count);
+	return 0;
+}
+
+// remove_area(id)
+int LuaAreaStore::l_remove_area(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+
+	LuaAreaStore *o = checkobject(L, 1);
+	AreaStore *ast = o->as;
+
+	u32 id = lua_tonumber(L, 2);
+	bool success = ast->removeArea(id);
+
+	lua_pushboolean(L, success);
+	return 1;
+}
+
+// set_cache_params(params)
+int LuaAreaStore::l_set_cache_params(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+
+	LuaAreaStore *o = checkobject(L, 1);
+	AreaStore *ast = o->as;
+
+	luaL_checktype(L, 2, LUA_TTABLE);
+
+	bool enabled = getboolfield_default(L, 2, "enabled", true);
+	u8 block_radius = getintfield_default(L, 2, "block_radius", 64);
+	size_t limit = getintfield_default(L, 2, "block_radius", 1000);
+
+	ast->setCacheParams(enabled, block_radius, limit);
+
+	return 0;
+}
+
+/* // to_string()
```

Comment:
Maybe you should deactivate this code with #if 0 instead (to not interfere with other comment blocks a developer working on this piece of code might add, and also to preserve syntax highlighting).

Also, why not add a comment talking about the problem with these APIs?  Maybe even a "TODO" comment to fix it later?